synapse/rest/media/v1/thumbnail_resource.py

```diff
@@ -384,8 +399,30 @@ async def _select_and_respond_with_thumbnail(
                 file_info.thumbnail.length,
             )
         else:
+            # This might be because:
+            # 1. We can't create thumbnails for the given media (corrupted or
+            #    unsupported file type)
+            # 2. The thumbnailing process never ran or errored out initially
+            #    when the media was first uploaded (these bugs should be
+            #    reported and fixed).
+            # 3. `dynamic_thumbnails` is disabled (see `homeserver.yaml`) and we
+            #    can't generate one. If this was enabled, Synapse would
+            #    go down a different code path.
             logger.info("Failed to find any generated thumbnails")
-            respond_404(request)
+
+            respond_with_json(
+                request,
+                400,
+                cs_error(
+                    "Cannot find any thumbnails for the requested media (%r). This might mean the media is not a supported_media_format=(%s) or your media is corrupted and cannot be thumbnailed. Additionally, dynamic thumbnails are disabled on this server."
```

Comment:
```suggestion
                    "Cannot find any thumbnails for the requested media (%r). This might mean the media is not a supported_media_format=(%s) or that thumbnailing failed for some outer reason. (Dynamic thumbnails are disabled on this server.)"
```